### PR TITLE
[Pool Simulator] Add warning alert when pool types don't match

### PR DESCRIPTION
**Pull request by yvesfracari**
Pool types considered stable={Fx, MetaStable, GyroE}
Pool types considered non stable={Gyro2, Gyro3}

Sometimes Gyro2/Gyro3 are used for stable assets, but this is not always true. I think this is the best way to deal with it, but I am open to new suggestions.

image

### CODE REVIEW

File: apps/balancer-tools/src/app/poolsimulator/(components)/AnalysisPage.tsx [hunk 1]
```diff
@@ -27,10 +35,21 @@ export default function Page() {
...
+  const poolsHaveDifferentPurpose =
+    poolTypes.filter((type) => stablePoolTypes.includes(type)).length == 1;
+
   return (
     <>
       <div className="flex lg:max-h-[calc(100vh-132px)] w-full flex-col gap-y-20 lg:overflow-auto pr-8 pt-8">
         {/* (h-screen - (header's height + footer's height)) = graph's height space */}
+        {poolsHaveDifferentPurpose && (
+          <AlertCard
+            alertType="warning"
```
Comment (ribeirojose) [inline apps/balancer-tools/src/app/poolsimulator/(components)/AnalysisPage.tsx:48]:
```suggestion
            style="warning"
```

File: apps/balancer-tools/src/app/poolsimulator/(components)/AnalysisPage.tsx [hunk 2]
```diff
@@ -27,10 +35,21 @@ export default function Page() {
     setCurrentTabTokenByIndex(tokensSymbol.indexOf(target.innerText));
   }
 
+  const poolTypes = [initialData?.poolType, customData?.poolType];
+  const poolsHaveDifferentPurpose =
```
Comment (ribeirojose) [inline apps/balancer-tools/src/app/poolsimulator/(components)/AnalysisPage.tsx:39]: `#note` if 0 or 2 match, this list won't have length 1 👍

File: apps/balancer-tools/src/components/AlertCard.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      messageBorderColor = "border-red-400";
+      messageTextColor = "text-tomato7";
+      break;
+    case "warning":
+      titleBgColor = "bg-amber9";
+      messageBgColor = "bg-amber12";
+      messageBorderColor = "border-yellow-300";
+      messageTextColor = "text-yellow-800";
+      break;
+  }
```
Comment (ribeirojose) [inline apps/balancer-tools/src/components/AlertCard.tsx:30]: would've just added the styles to the `cn` call to make it simpler

File: apps/balancer-tools/src/components/AlertCard.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+  return (
+    <div role="alert">
+      <div
+        className={cn(
+          "font-bold text-slate12 rounded-t px-4 py-2 mt-1",
+          titleBgColor,
```
Comment (ribeirojose) [inline apps/balancer-tools/src/components/AlertCard.tsx:37]:
```suggestion
          {"bg-tomato9": alertType === "error"},
          {"bg-amber9": alertType === "warning"},
          ...
```

**Review by ribeirojose — APPROVED** [review]
LGTM 👍

File: apps/balancer-tools/src/app/poolsimulator/(components)/AnalysisPage.tsx (hunk 2 above)
Comment (yvesfracari) [reply apps/balancer-tools/src/app/poolsimulator/(components)/AnalysisPage.tsx:39]: I can leave a comment about this, but actually that is what we want. If 0 or 2 match is bc none matches is bc both are for non-stable and if both matches is bc both are stable.

Comment (ribeirojose) [reply apps/balancer-tools/src/app/poolsimulator/(components)/AnalysisPage.tsx:39]: Yea I understood it, just left a note to self/others reviewing the PR 👍

**Pull request merged**